assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -265,11 +265,13 @@ const AddressForm = ( {
 							} )
 						}
 						customValidation={ ( inputObject: HTMLInputElement ) =>
-							customValidationHandler(
-								inputObject,
-								field.key,
-								values
-							)
+							field.required || inputObject.value
+								? customValidationHandler(
+										inputObject,
```

Comment:
I think these checks should be inside the customValidationHandler if possible. The `customValidationHandler` should receive another parameter `isOptional`, defaulted to false and should handle the logic inside.